src/ResourceManager/Consumption/Commands.Consumption/Cmdlets/Budget/RemoveAzureRmConsumptionBudget.cs

```diff
@@ -0,0 +1,66 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Consumption.Common;
+using Microsoft.Azure.Management.Consumption;
+using Microsoft.Azure.Management.Consumption.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Consumption.Cmdlets.Budget
+{
+    [Cmdlet(VerbsCommon.Remove, "AzureRmConsumptionBudget", SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    public class RemoveAzureRmConsumptionBudget : AzureConsumptionCmdletBase
+    {
```

Comment:
@maddieclayton We don't have piping implemented in budget, as we have budget name as the key instead of budget object. Remove or Set cmdlet asks for same parameter as Get cmdlet, and piping in Remove seems redundant in this scenario. Lets know if piping is required for all CRUD cmdlets

Reply:
@bgsky Does the budget object have the budget name as a property? If this is the case, when you implement the InputObject parameter set, you will just extract the budget name from the object.